lib/src/effective_text.dart

```diff
@@ -0,0 +1,71 @@
+// ignore_for_file: require_trailing_commas
+
+import 'package:dartx/dartx.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:spot/spot.dart';
+import 'package:spot/src/checks/context.dart';
+
+/// Matchers for [Text] auto-generated by spot
+///
+/// ```dart
+/// spot<Text>().printMatchers(propNameOverrides: (MapEntry('data': 'text')));
+/// ```
+extension EffectiveTextMatcher on WidgetMatcher<Text> {
+  WidgetMatcher<Text> hasEffectiveMaxLinesWhere(MatchProp<int> match) {
+    return hasProp<int>('maxLines', match);
```

Comment:
The current implementation is `hasMaxLinesWhere`, not `hasEffectiveMaxLinesWhere`
```suggestion
  WidgetMatcher<Text> hasEffectiveMaxLinesWhere(MatchProp<int?> match) {
       final richTextElement =
        element.children.firstWhere((e) => e.widget is RichText);
    final richText = richTextElement.widget as RichText;
    final actual = richText.maxLines;

    final ConditionSubject<int?> conditionSubject = it<int?>();
    final Subject<int?> subject = conditionSubject.context.nest<int?>(
      () => ['with prop "maxLines"'],
      (value) => Extracted.value(actual),
    );
    match(subject);
    final failure = softCheck(actual, conditionSubject);
    if (failure != null) {
      final errorMessage =
          describe(conditionSubject).map((it) => it.trim()).toList().join(' ');
      throw TestFailure(
        'Failed to match widget: $errorMessage, actual: ${literal(actual).joinToString()}',
      );
    }
    return this;
```